etna/transforms/math/binary_operator.py

```diff
@@ -0,0 +1,181 @@
+        self.operator = BinaryOperator(operator)
+        self.out_column = out_column if out_column is not None else self.left_column + self.operator + self.right_column
+
+        self._in_column_regressor: Optional[bool] = None
+        self.inverse_operator = BinaryOperator(inverse_logic[operator]) if operator in inverse_logic else None
+
+    def fit(self, ts: TSDataset) -> "BinaryOperationTransform":
+        """Fit the transform."""
+        self._in_column_regressor = self.left_column in ts.regressors and self.right_column in ts.regressors
+        super().fit(ts)
+        return self
+
+    def _fit(self, df: pd.DataFrame) -> "BinaryOperationTransform":
+        """Fit preprocess method, does nothing in ``BinaryOperationTransform`` case.
+
+        Parameters
+        ----------
+        df:
+            dataframe with data.
+
+        Returns
+        -------
+        result: ``BinaryOperationTransform``
+        """
+        return self
+
+    def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Perform operation on passed dataframe.
+
+        Parameters
+        ----------
+        df:
+            dataframe with data to transform.
+
+        Returns
+        -------
+        : pd.Dataframe
+            transformed dataframe
+        """
+        result = self.operator.perform(
+            df=df,
+            left_operand=self.left_column,
+            right_operand=self.right_column,
+            out_column=self.out_column,
+        )
+        if self.inplace_flag:
+            df.loc[:, pd.IndexSlice[:, self.out_column]] = result
+        else:
+            df = pd.concat((df, result), axis=1)
+        return df
+
+    def _inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Perform reverse operation on passed dataframe.
+        If
+
+        Parameters
+        ----------
+        df:
+            dataframe with data to transform.
+
+        Returns
+        -------
+        : pd.Dataframe
+            transformed dataframe
+        Raises
+        ------
+        ValueError:
+            if out_column is not left_column or right_column
+        ValueError:
+            If initial operation is not '+', '-', '*' or '/'
+        """
+        if not self.inplace_flag or self.inverse_operator is None:
+            raise ValueError("We only support inverse transform if out_column is left_column or right_column")
+        if self.inverse_operator is None:
+            raise ValueError("We only support inverse transform if the original operation is '+', '-', '*', '/'")
+        proper_column = self.left_column if (self.left_column != self.out_column) else self.right_column
+        if self.operator in ["+", "*"]:
+            df.loc[:, pd.IndexSlice[:, self.out_column]] = self.inverse_operator.perform(
+                df=df, left_operand=self.out_column, right_operand=proper_column, out_column=self.out_column
+            )
+        else:
```

Comment:
It looks difficult to understand...
This logic that inverse operator for + is always - should probably be reconsidered, because it just isn't true and confuses.